### PR TITLE
Reduce duplication in codebase

### PR DESCRIPTION
**Pull request by mjgiarlo**

### CODE REVIEW

File: app/controllers/sufia/resource_sync_controller.rb [hunk 1]
```diff
@@ -1,26 +1,14 @@
...
-    body = Rails.cache.fetch("source_description_#{request.host}", expires_in: 1.week) do
-      build_source_description
-    end
-    render body: body, content_type: 'application/xml'
+    render_from_cache_as_xml(build_source_description)
   end
 
   def capability_list
-    # Caching based on host, for multitenancy support
-    body = Rails.cache.fetch("source_description_#{request.host}", expires_in: 1.week) do
```
Comment (jcoyne) [inline app/controllers/sufia/resource_sync_controller.rb]: These should have had different cache keys.  This is a bug.

File: app/controllers/sufia/resource_sync_controller.rb [hunk 2]
```diff
@@ -1,26 +1,14 @@
...
-      build_capability_list
-    end
-    render body: body, content_type: 'application/xml'
+    render_from_cache_as_xml(build_capability_list)
   end
 
   def resource_list
-    # Caching based on host, for multitenancy support
-    body = Rails.cache.fetch("source_description_#{request.host}", expires_in: 1.week) do
```
Comment (jcoyne) [inline app/controllers/sufia/resource_sync_controller.rb]: Here too.

File: app/controllers/sufia/resource_sync_controller.rb [hunk 3]
```diff
@@ -1,26 +1,14 @@
 class Sufia::ResourceSyncController < ApplicationController
   def source_description
-    # Caching based on host, for multitenancy support
-    body = Rails.cache.fetch("source_description_#{request.host}", expires_in: 1.week) do
-      build_source_description
-    end
-    render body: body, content_type: 'application/xml'
+    render_from_cache_as_xml(build_source_description)
```
Comment (jcoyne) [inline app/controllers/sufia/resource_sync_controller.rb]: This is obviating the cache.  It's running the build_source_description even if it's available in the cache.

File: app/controllers/sufia/resource_sync_controller.rb [hunk 4]
```diff
@@ -1,26 +1,14 @@
...
   end
 
   def capability_list
-    # Caching based on host, for multitenancy support
-    body = Rails.cache.fetch("source_description_#{request.host}", expires_in: 1.week) do
-      build_capability_list
-    end
-    render body: body, content_type: 'application/xml'
+    render_from_cache_as_xml(build_capability_list)
```
Comment (jcoyne) [inline app/controllers/sufia/resource_sync_controller.rb]: It's not being cached if you call it here.

File: app/controllers/sufia/resource_sync_controller.rb [hunk 5]
```diff
@@ -1,26 +1,14 @@
...
   end
 
   def resource_list
-    # Caching based on host, for multitenancy support
-    body = Rails.cache.fetch("source_description_#{request.host}", expires_in: 1.week) do
-      build_resource_list
-    end
-    render body: body, content_type: 'application/xml'
+    render_from_cache_as_xml(build_resource_list)
```
Comment (jcoyne) [inline app/controllers/sufia/resource_sync_controller.rb]: and not cached here.

File: app/controllers/sufia/resource_sync_controller.rb (hunk 1 above)
Comment (mjgiarlo) [inline app/controllers/sufia/resource_sync_controller.rb]: Done below.

File: app/controllers/sufia/resource_sync_controller.rb (hunk 2 above)
Comment (mjgiarlo) [inline app/controllers/sufia/resource_sync_controller.rb]: Done.

**Pull request merged**